pandas/core/series.py

```diff
@@ -4564,6 +4565,96 @@ def replace(
+            >>> text_values = ['alpha', 'beta', 'gamma', 'delta', 'epsilon']
+            >>> s = pd.Series(text_values, index=int_values)
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            Series name: None
+            Non-Null Count  Dtype
+            --------------  -----
+            5 non-null      object
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Prints a summary excluding information about its values:
+
+            >>> s.info(verbose=False)
+            <class 'pandas.core.series.Series'>
+            Int64Index: 5 entries, 1 to 5
+            dtypes: object(1)
+            memory usage: 80.0+ bytes
+
+            Pipe output of Series.info to buffer instead of sys.stdout, get
+            buffer content and writes to a text file:
+
+            >>> import io
+            >>> buffer = io.StringIO()
+            >>> s.info(buf=buffer)
+            >>> s = buffer.getvalue()
+            >>> with open("df_info.txt", "w",
+            ...           encoding="utf-8") as f:  # doctest: +SKIP
+            ...     f.write(s)
+            260
+
+            The `memory_usage` parameter allows deep introspection mode, specially
+            useful for big Series and fine-tune memory optimization:
+
+            >>> random_strings_array = np.random.choice(['a', 'b', 'c'], 10 ** 6)
+            >>> s = pd.Series(np.random.choice(['a', 'b', 'c'], 10 ** 6))
+            >>> s.info()
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 7.6+ MB
+
+            >>> s.info(memory_usage='deep')
+            <class 'pandas.core.series.Series'>
+            RangeIndex: 1000000 entries, 0 to 999999
+            Series name: None
+            Non-Null Count    Dtype
+            --------------    -----
+            1000000 non-null  object
+            dtypes: object(1)
+            memory usage: 55.3 MB"""
+        ),
+        see_also_sub=(
+            """
+            Series.describe: Generate descriptive statistics of Series.
+            Series.memory_usage: Memory usage of Series."""
+        ),
+    )
+    @doc(SeriesInfo.to_buffer)
+    def info(
+        self,
+        verbose: Optional[bool] = None,
+        buf: Optional[IO[str]] = None,
+        max_cols: Optional[int] = None,
+        memory_usage: Optional[Union[bool, str]] = None,
+        null_counts: Optional[bool] = None,
+    ) -> None:
+        if max_cols is not None:
+            raise ValueError(
+                "Argument `max_cols` can only be passed "
+                "in DataFrame.info, not Series.info"
+            )
+        return SeriesInfo(self, memory_usage).to_buffer(
```

Comment:
it seems odd imo to have parameters other than buf passed to to_buffer()

would it be better to pass verbose and show_counts to SeriesInfo constructor or rename to_buffer?

Reply:
Putting the params in the constructor is possible, but in this case in ``SeriesInfo`` there will be two more attributes, which are used only in one method (smaller cohesion within the class).
I would prefer renaming the method. I will look into that.

Reply:
I renamed ``to_buffer`` -> ``render``.

However, I had to make the same function signature for ``DataFrameInfo`` and ``SeriesInfo`` to avoid typing errors.
Thus, I pass ``max_cols`` into ``render`` and raise ValueError there instead of ``pandas.core.series.info``.
How does it look?